src/main/java/uk/ac/ebi/subs/validator/data/FileUploadValidationMessageEnvelope.java

```diff
@@ -0,0 +1,71 @@
+package uk.ac.ebi.subs.validator.data;
+
+import uk.ac.ebi.subs.data.fileupload.File;
+import uk.ac.ebi.subs.data.submittable.BaseSubmittable;
+import uk.ac.ebi.subs.data.submittable.Submittable;
+
+import java.util.Collections;
+import java.util.stream.Stream;
+
+/**
+ * This is a Data Transfer Object transferring data from the {@code file-upload-service} service
+ * to the file reference validator.
+ * A container object holding the UUID (MongoDB ID) and the file to validate.
+ *
+ * Created by karoly on 07/04/2018.
+ */
+public class FileUploadValidationMessageEnvelope {
+
+    private String validationResultUUID;
+    private int validationResultVersion;
+    private File fileToValidate;
+    private String submissionId;
+
+    public FileUploadValidationMessageEnvelope() {
+    }
+
+    public FileUploadValidationMessageEnvelope(String validationResultUUID, int validationResultVersion, File fileToValidate) {
+        this.validationResultUUID = validationResultUUID;
+        this.validationResultVersion = validationResultVersion;
+        this.fileToValidate = fileToValidate;
+    }
+
+    public FileUploadValidationMessageEnvelope(String validationResultUUID, int validationResultVersion, File fileToValidate, String submissionId) {
```

Comment:
In which scenario do we need a constructor with and other without the `submissionId` ?

If we need both, we can use an overload method avoiding the boilerplate of setting all the properties repeatedly:
```java
public FileUploadValidationMessageEnvelope() {}

public FileUploadValidationMessageEnvelope(String validationResultUUID, int validationResultVersion, File fileToValidate) {
  this(validationResultUUID, validationResultVersion, fileToValidate, null); 
}

public FileUploadValidationMessageEnvelope(String validationResultUUID, int validationResultVersion, File fileToValidate, String submissionId) {
  this.validationResultUUID = validationResultUUID;
  this.validationResultVersion = validationResultVersion;
  this.fileToValidate = fileToValidate;
  this.submissionId = submissionId;
}
```